addons/udes_simple_location_blocking/models/stock_location.py

```diff
@@ -0,0 +1,75 @@
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class StockLocation(models.Model):
+    _inherit = "stock.location"
+
+    u_blocked = fields.Boolean(
+        string="Is Blocked?",
+        help=_("Check this box to prevent stock picks from this location"),
+        index=True,
+    )
+
+    u_blocked_reason = fields.Char(string="Reason for Block:", required=False)
+
+    def check_blocked(self, prefix=""):
+        """
+        Checks if any of the locations in self are blocked and if blocked raise a ValidationError.
+
+        :param prefix: string
+        """
+        if not isinstance(prefix, str):
+            raise ValidationError(_("Prefix parameter for check_blocked should be string"))
+        blocked_locations = self.filtered(lambda x: x.u_blocked)
+        if blocked_locations:
+            raise ValidationError(
+                _("%s %s Please speak to a team leader to resolve the issue.")
+                % (prefix, "".join(blocked_locations._prepare_blocked_msg()))
+            )
+
+    def _prepare_blocked_msg(self):
+        """
+        Prepares a message for the locations depending on if it is blocked or not.
+        """
+        msg = []
+        for location in self:
+            if location.u_blocked:
+                reason = _("(reason: no reason specified)")
+                if location.u_blocked_reason:
+                    reason = _("(reason: %s)") % (str(location.u_blocked_reason))
+                msg.append(_("Location %s is blocked %s.") % (location.name, reason))
+            else:
+                msg.append(_("Location %s is not blocked.") % location.name)
+        msg = " ".join(msg)
+        return msg
+
+    @api.onchange("u_blocked")
+    def onchange_u_blocked(self):
+        """Empty blocked reason when locations are unblocked"""
+        if not self.u_blocked:
```

Comment:
I always forget - is self always singleton here? Or should we loop?

Reply:
I looked at the api.py but it doesn't specify. I would assume since this is called when the record is modified on the GUI that it returns a single record? I don't think you can modify a field for multiple records at once in the GUI.

Reply:
I checked with Armand and he also thinks that it's always a singleton but he didn't say for sure. I cannot find a definitive answer.

Reply:
haven't looked, do we have an option to unblock multiple locations at once? With the select, and unblock or is it just going into each location (or parent loc) and unblocking it?

Reply:
There is a field on the form for each location. I cannot see an additional button to unblock multiple locations at once.